challenge/contact.py

```diff
@@ -1,188 +0,0 @@
-#!/usr/bin/env python
-#
-from selenium import webdriver
-from urllib.parse import urlparse,urlunparse
-import urllib.request
-
-import logging
-import pdb
-import re
-import sys
-import platform
-
-class ui:
-    """ web access with ui capabilities """
-    def __init__(self,hack=''):
-        """ Selenium web driver can be set up 'headless' browser on some remote system, which could improve performance or allow running multiple copies"""
-        if hack:
-            self.web = webdriver.Firefox(webdriver.FirefoxProfile(hack))
-        else:
-            self.web = webdriver.Firefox()
-
-        self.actions = webdriver.common.action_chains.ActionChains(self.web)
-        #self.scan.actions.getCurrentWindow().focus()
-        for _ in range(12):
-            self.actions.send_keys(webdriver.common.keys.Keys.TAB)
-        #for _ in range(3)
-        #    self.actions.send_keys(webdriver.common.keys.Keys.PAGE_DOWN)
-
-        # perform actions later
-    def __enter__(self):
-        return self
-    def __exit__(self,t,v,s):
-        self.web.quit()
-
-scheduled = set() # deque? avoid duplicate scheduling
-done = set() # web pages already examined
-emails = set() # email addresses found
-
-def normalized(link):
-    url = urlparse(link)
-    urlparts = list(url)
-    if urlparts[0] == 'http' or urlparts[0] == '':
-        urlparts[0] = 'http'
-    urlparts[0] = str.lower(urlparts[0]) # scheme
-    urlparts[1] = str.lower(urlparts[1]) # netloc
-    if 'mailto' == urlparts[0]:
-        urlparts[2] = str.lower(urlparts[2]) # email address in path
-    urlparts[3] = '' # parms
-    urlparts[4] = '' # query
-    urlparts[5] = '' # fragment
-    return urlunparse(urlparts)
-
-class Links:
-    """ iterator for links on a web page """
-    def __str__(self):
-        return str({'domain': self.domain,'page':self.page,'scheduled':self.found,'emails':self.mail})
-
-    def __init__(self,scan,page,domain):
-        self.scan = scan 
-        self.page = page
-        self.domain = domain
-        self.mail = set()
-        try:
-            #scan.web.manage().timeouts().implicitlyWait(10,TimeUnit.SECONDS)
-            self.scan.web.get(page)
-            # TODO: deal with model dialogs
-            # TODO: check if page really html?
-            # TODO: set focus on page first, before any key commands
-            scan.actions.perform()
-            #pdb.set_trace()
-            aa = '//a[@href]'
-            aas = self.scan.web.find_elements_by_xpath(aa)
-            href = [normalized(a.get_attribute('href')) for a in aas]
-            # links with domain
-            lsd = set([h for h in href if self.domain in h])
```

Comment:
wouldn't this incorrectly return True if self.domain was "foo.com" but the url in the href was "foo.company.com" ?